packages/graph-explorer/src/modules/GraphViewer/useGraphStyles.ts

```diff
@@ -1,120 +1,107 @@
   getEdgeIdFromRenderedEdgeId,
   RenderedEdgeId,
   useDisplayEdgesInCanvas,
+  VertexTypeConfig,
 } from "@/core";
 import type { GraphProps } from "@/components";
-import useTextTransform from "@/hooks/useTextTransform";
-import { renderNode } from "./renderNode";
 import {
   useEdgeTypeConfigs,
   useVertexTypeConfigs,
 } from "@/core/ConfigurationProvider/useConfiguration";
 import { MISSING_DISPLAY_VALUE } from "@/utils/constants";
-import { useQueryClient } from "@tanstack/react-query";
+import { useBackgroundImageMap } from "./useBackgroundImageMap";
 
 const LINE_PATTERN = {
   solid: undefined,
   dashed: [5, 6],
   dotted: [1, 2],
 };
 
-const useGraphStyles = () => {
+export default function useGraphStyles() {
   const vtConfigs = useVertexTypeConfigs();
   const etConfigs = useEdgeTypeConfigs();
-  const textTransform = useTextTransform();
-  const [styles, setStyles] = useState<GraphProps["styles"]>({});
   const displayEdges = useDisplayEdgesInCanvas();
-  const client = useQueryClient();
 
   const deferredVtConfigs = useDeferredValue(vtConfigs);
   const deferredEtConfigs = useDeferredValue(etConfigs);
 
-  useEffect(() => {
-    (async () => {
-      const styles: GraphProps["styles"] = {};
+  const backgroundImageMap = useBackgroundImageMap(deferredVtConfigs);
 
-      for (const vtConfig of deferredVtConfigs) {
-        const vt = vtConfig.type;
-
-        // Process the image data or SVG
-        const backgroundImage = await renderNode(client, vtConfig);
-
-        styles[`node[type="${vt}"]`] = {
-          "background-image": backgroundImage,
-          "background-color": vtConfig.color,
-          "background-opacity": vtConfig.backgroundOpacity,
-          "border-color": vtConfig.borderColor,
-          "border-width": vtConfig.borderWidth,
-          "border-opacity": vtConfig.borderWidth ? 1 : 0,
-          "border-style": vtConfig.borderStyle,
-          shape: vtConfig.shape,
-          width: 24,
-          height: 24,
-        };
-      }
+  return createGraphStyles(
+    deferredVtConfigs,
+    deferredEtConfigs,
+    backgroundImageMap,
+    displayEdges
+  );
+}
 
-      for (const etConfig of deferredEtConfigs) {
-        const et = etConfig?.type;
+function createGraphStyles(
+  deferredVtConfigs: VertexTypeConfig[],
+  deferredEtConfigs: EdgeTypeConfig[],
+  backgroundImageMap: Map<string, string | null>,
+  displayEdges: Map<EdgeId, DisplayEdge>
+): GraphProps["styles"] {
+  const styles: GraphProps["styles"] = {};
 
-        let label = textTransform(et);
-        if (label.length > 20) {
-          label = label.substring(0, 17) + "...";
-        }
+  for (const vtConfig of deferredVtConfigs) {
+    const vt = vtConfig.type;
 
-        styles[`edge[type="${et}"]`] = {
-          label,
-          "source-distance-from-node": 0,
-          "target-distance-from-node": 0,
-        };
+    // Process the image data or SVG
+    const backgroundImage = backgroundImageMap.get(vt) ?? undefined;
```

Comment:
Map returns `string | null` but converts to `undefined`.  Consider defaulting to `null` instead.

Reply:
The result is heading to Cytoscape, which defines the type as `string | undefined`.